src/macaron/slsa_analyzer/provenance/loader.py

```diff
@@ -1,48 +1,75 @@
-# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 - 2024, Oracle and/or its affiliates. All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 """This module contains the loaders for SLSA provenances."""
 
 import base64
+import configparser
 import gzip
+import io
 import json
 import zlib
+from urllib.parse import urlparse
 
+import requests
+
+from macaron.config.defaults import defaults
 from macaron.slsa_analyzer.provenance.intoto import InTotoPayload, validate_intoto_payload
 from macaron.slsa_analyzer.provenance.intoto.errors import LoadIntotoAttestationError, ValidateInTotoPayloadError
 from macaron.util import JsonType
 
 
-def load_provenance_file(filepath: str) -> dict[str, JsonType]:
-    """Load a provenance file and obtain the payload.
+def _try_read_url_link_file(file_content: bytes) -> str | None:
+    parser = configparser.ConfigParser()
+    try:
+        parser.read_file(io.TextIOWrapper(io.BytesIO(file_content), encoding="utf-8"))
```

Comment:
Hmm, I assume we can use [`parser.read_string()`](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.read_string) instead since we have `file_content` as input instead of a file object?
```suggestion
        parser.read_string(file_content.decode())
```
The [`bytes.decode()`](https://docs.python.org/3/library/stdtypes.html#bytes.decode) method converts `bytes` to `str` with `utf-8` as the default encoding.

Reply:
I didn't know about `bytes.decode()`, that is much simpler. Done.